soot-infoflow-android/src/soot/jimple/infoflow/android/entryPointCreators/AndroidEntryPointConstants.java

```diff
@@ -125,8 +125,8 @@
 
 	private static final String[] activityMethods = { ACTIVITY_ONCREATE, ACTIVITY_ONDESTROY, ACTIVITY_ONPAUSE,
 			ACTIVITY_ONRESTART, ACTIVITY_ONRESUME, ACTIVITY_ONSTART, ACTIVITY_ONSTOP, ACTIVITY_ONSAVEINSTANCESTATE,
-			ACTIVITY_ONRESTOREINSTANCESTATE, ACTIVITY_ONCREATEDESCRIPTION, ACTIVITY_ONPOSTCREATE,
-			ACTIVITY_ONPOSTRESUME };
+			ACTIVITY_ONRESTOREINSTANCESTATE, ACTIVITY_ONCREATEDESCRIPTION, ACTIVITY_ONPOSTCREATE, ACTIVITY_ONPOSTRESUME,
+			ACTIVITY_ONATTACHFRAGMENT };
```

Comment:
Das ``onAttachFragment`` sollte explizit behandelt werden, weil hier das korrekte Fragment übergeben werden soll an Stelle eines generischen Werts (i.d.R. ``null``). Bitte schau noch einmal in den Code und prüfe, dass es wirklich eingefügt wird (dort, wo die Fragments instantiiert werden). Dann muss es hier wieder raus.

Reply:
Das ist korrekt, `ACTIVITY_ONATTACHFRAGMENT` wird an anderer Stelle genutzt als die anderen Konstanten `ACTIVITY_ON...`. Die `activityMethodList` wird aber dafür nicht genutzt, sondern wird nur in [`getLifecycleMethods()`](https://github.com/secure-software-engineering/FlowDroid/blob/1d26d9ce5b5a104b5f8132c459b1f2e96674be0d/soot-infoflow-android/src/soot/jimple/infoflow/android/callbacks/DefaultCallbackAnalyzer.java#L175) und [isEntryPointMethod()](https://github.com/secure-software-engineering/FlowDroid/blob/1d26d9ce5b5a104b5f8132c459b1f2e96674be0d/soot-infoflow-android/src/soot/jimple/infoflow/android/entryPointCreators/AndroidEntryPointUtils.java#L153) verwendet. Wenn ich das richtig sehe, werden diese Methoden (zumindest teilweise) verwendet, um zu prüfen, welche Methoden grundsätzlich Lifecycle-Methoden sind. Sollte dort `onAttachFragment` dann auch bewusst nicht dabei sein?